docs/aws/audit/elbmonitoring/rules/elb_web_tier_security_policy.mdx

```diff
@@ -23,6 +23,189 @@ HIPAA, GDPR, NIST, AWSWAF
+4. Check the SSL Certificate column. If the column is empty or the policy is not up-to-date, then the ELB is not using the latest AWS security policy for SSL negotiations.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS CLI website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Load Balancers: Use the following command to list all the load balancers in your account.
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+   This command will return a JSON object with details about all your load balancers.
+
+3. Check the SSL Policies: For each load balancer, check the SSL policies by using the following command:
+
+   ```
+   aws elbv2 describe-ssl-policies --names [policy-name]
+   ```
+
+   Replace `[policy-name]` with the name of the SSL policy you want to check. This command will return a JSON object with details about the specified SSL policy.
+
+4. Analyze the output: Look for the "SslProtocols" field in the output. If the policy includes outdated or insecure protocols (like SSLv3 or TLS 1.0), then the load balancer is not using the latest AWS security policy for SSL negotiations.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Now, create a client for ELB (Elastic Load Balancer) and retrieve all the load balancers.
+
+```python
+elb = session.client('elbv2')
+load_balancers = elb.describe_load_balancers()
+```
+
+4. Iterate over each load balancer and check the SSL policy. If the policy is not the latest, print a message.
+
+```python
+for lb in load_balancers['LoadBalancers']:
+    listeners = elb.describe_listeners(LoadBalancerArn=lb['LoadBalancerArn'])
+    for listener in listeners['Listeners']:
+        if 'SslPolicy' in listener and listener['SslPolicy'] != 'ELBSecurityPolicy-2016-08':
+            print(f"Load balancer {lb['LoadBalancerName']} is using outdated SSL policy {listener['SslPolicy']}")
+```
+
+This script will print out the names of all load balancers that are using an SSL policy other than 'ELBSecurityPolicy-2016-08', which is the latest policy. Please replace 'YOUR_ACCESS_KEY', 'YOUR_SECRET_KEY', and 'YOUR_REGION' with your actual AWS credentials and region.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>